python/strings_udf/strings_udf/__init__.py

```diff
@@ -0,0 +1,41 @@
+# Copyright (c) 2022, NVIDIA CORPORATION.
+from ptxcompiler.patch import patch_needed, CMD
+import os
+import warnings
+import sys
+import subprocess
+
+
+def versions_compatible(path):
+    """
+    Example PTX header:
+
+    // Generated by NVIDIA NVVM Compiler
+    //
+    // Compiler Build ID: CL-30672275
+    // Cuda compilation tools, release 11.5, V11.5.119
+    // Based on NVVM 7
+    """
+    # obtain the cuda version used to compile this PTX file
+    file = open(path).read()
+    major, minor = file.split("\n")[4].split(" ")[-2][:-1].split(".")
```

Comment:
I'm hesitant about hard coding the indices here. Any changes to the end of that line could break this. Can we do a regex search for something like `Cuda compilation tools, release ([0-9\.]+),` and use the captured group?

Reply:
Updated this to search using `re`.